collated/test/A0132157M reused.md

```diff
@@ -0,0 +1,2006 @@
+    private static final String task_LIST_VIEW_ID = "#deadlineListView";
+
+    public DeadlineListPanelHandle(GuiRobot guiRobot, Stage primaryStage) {
+        super(guiRobot, primaryStage, TestApp.APP_TITLE);
+    }
+
+    public List<ReadOnlyTask> getSelectedtasks() {
+        ListView<ReadOnlyTask> taskList = getListView();
+        return taskList.getSelectionModel().getSelectedItems();
+    }
+
+    public ListView<ReadOnlyTask> getListView() {
+        return (ListView<ReadOnlyTask>) getNode(task_LIST_VIEW_ID);
+    }
+
+    /**
+     * Returns true if the list is showing the task details correctly and in correct order.
+     * @param tasks A list of task in the correct order.
+     */
+    public boolean isListMatching(ReadOnlyTask... tasks) {
+        return this.isListMatching(0, tasks); //something wrong, always return false!!!
+    }
+    
+    /**
+     * Clicks on the ListView.
+     */
+    public void clickOnListView() {
+        Point2D point= TestUtil.getScreenMidPoint(getListView());
+        guiRobot.clickOn(point.getX(), point.getY());
+    }
+
+    /**
+     * Returns true if the {@code tasks} appear as the sub list (in that order) at position {@code startPosition}.
+     */
+    public boolean containsInOrder(int startPosition, ReadOnlyTask... tasks) {
+        List<ReadOnlyTask> tasksInList = getListView().getItems();
+
+        // Return false if the list in panel is too short to contain the given list
+        if (startPosition + tasks.length > tasksInList.size()){
+            return false;
+        }
+
+        // Return false if any of the tasks doesn't match
+        for (int i = 0; i < tasks.length; i++) {
+            if (!tasksInList.get(startPosition + i).getName().name.equals(tasks[i].getName().name)){
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns true if the list is showing the task details correctly and in correct order.
+     * @param startPosition The starting position of the sub list.
+     * @param tasks A list of task in the correct order.
+     */
+    public boolean isListMatching(int startPosition, ReadOnlyTask... tasks) throws IllegalArgumentException {
+        if (tasks.length + startPosition != getListView().getItems().size()) {
+            throw new IllegalArgumentException("List size mismatched\n" +
+                    "Expected " + (getListView().getItems().size() - 1) + " tasks");
+        }
+        assertTrue(this.containsInOrder(startPosition, tasks));
+        for (int i = 0; i < tasks.length; i++) {
+            final int scrollTo = i + startPosition;
+            guiRobot.interact(() -> getListView().scrollTo(scrollTo));
+            guiRobot.sleep(200);
+            if (!TestUtil.compareCardAndDeadline(getDeadlineCardHandle(startPosition + i), tasks[i])) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+
+    public DeadlineCardHandle navigateToDeadline(String readOnlyTask) {
```

Comment:
you can claim this test as authored by you! (and other similar tests, that involve non-trivial changes)
please refer to one of the recent announcements by Prof. Damith